lib/src/lifecycle_module.dart

```diff
@@ -17,25 +17,55 @@ library w_module.src.lifecycle_module;
 import 'dart:async';
 
 import 'package:logging/logging.dart';
-import 'package:meta/meta.dart' show protected;
+import 'package:meta/meta.dart' show protected, required;
 import 'package:w_common/disposable.dart';
 
+/// Possible states a [LifecycleModule] may occupy.
+enum LifecycleState {
+  /// The module has been instantiated.
+  initialized,
```

Comment:
Maybe just change this to `instantiated` to make sure it isn't confused with whatever happens when you call an `init()` method on various modules.